incubator/sparkoperator/templates/spark-operator.yaml

```diff
@@ -0,0 +1,153 @@
+# If the admission webhook is enabled, then the following pre-install steps are performed
+# 1. Create the serviceAccount "sparkoperator"
+# 2. Create the namespace ("sparkoperator" by default)
+# 3. Generate and install the secret in the namespace created in step 2.
+
+# Steps above need to be executed in the order presented because the first two steps are the pre-requisites
+# for step 3 (e.g. In step 3, the token corresponding to the service account created in step 1 is used to communicate
+# with the Kubernetes API server to install the secret bundle. The order are specified using helm.sh/hook-weight
+# annotations.
+
+{{ if .Values.enableWebhook }}
```

Comment:
Just out of curiosity, what would happen if you just leave this block default for everything (with hook-weights), also for without webhooks?

Reply:
I could do that. But in that case, the user would need to manually delete the namespace when the chart release is deleted regardless of the setting of `enableWebhook`. The current template (i.e. with the `if` condition) saves the user from doing that when `enableWebhook` is `false`.